docker/start_postgres.sh

```diff
@@ -13,12 +15,25 @@ if [[ "$ACTION" == "start" ]]; then
     fi
 fi
 
+# Bootstrap the PGDATA folder
+if [[ "$ACTION" == "init" ]]; then
```

Comment:
Would it make sense to, instead of testing `init`, have the script check if `PGDATA` is provided and (if the action is `start`) initialize that instead? This way no additional manual init is required, and a simple change in the invocation of `docker run` is sufficient.

Reply:
If PGDATA is provided, `init` isn't needed. Perhaps you mount in a persistent volume that's already initialized, so it wouldn't need that setup step. The `init` is just a helper to bootstrap an empty folder (at $PGDATA) to make getting started easier.

Reply:
It felt a little creepy making it somehow detect if PGDATA was ready-to-roll and if not, automatically do it. I don't think I could anticipate all of the failure cases?

My workflow with PL dev is usually to run a shell in the container anyway and not the launch scripts. Since then I need to run `/PrairieLearn/docker/start_postgres.sh` first anyway, making the `init` command separate seemed reasonable.

Reply:
That actually makes sense. Long shot: what about running `pg_ctl status` and checking the status code? If the status code is 4, the directory is not valid and needs to be initialized (https://www.postgresql.org/docs/current/app-pg-ctl.html).

Reply:
I can noodle on that.

Reply:
OK, I have a version of the script that automatically initializes the database sbased on `pg_ctl status` as you described. I think I'm running into a new race condition with workspaces bootstrapping. I'll post about that in a new comment.